### PR TITLE
differential shooter -> master

### PR DESCRIPTION
**Pull request by NoamRosenberg08**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/subsystems/Shooter/FlyWheel/FlyWheel.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package edu.greenblitz.robotName.subsystems.shooter.FlyWheel;
+package edu.greenblitz.robotName.subsystems.Shooter.FlyWheel;
```
Comment (itamaroryan) [inline src/main/java/edu/greenblitz/robotName/subsystems/Shooter/FlyWheel/FlyWheel.java:1]: Package name convetntions!

File: src/main/java/edu/greenblitz/robotName/subsystems/Shooter/FlyWheel/FlyWheelConstants.java [hunk 2]
```diff
@@ -1,18 +1,19 @@
-package edu.greenblitz.robotName.subsystems.shooter.FlyWheel;
+package edu.greenblitz.robotName.subsystems.Shooter.FlyWheel;
```
Comment (itamaroryan) [inline src/main/java/edu/greenblitz/robotName/subsystems/Shooter/FlyWheel/FlyWheelConstants.java:1]: Package naming conventions!

File: src/main/java/edu/greenblitz/robotName/subsystems/Shooter/FlyWheel/NeoFlyWheel/NeoFlyWheel.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        );
+        leftMotor.getPIDController().setReference(
+                leftVelocity,
+                CANSparkMax.ControlType.kVelocity,
+                NeoFlyWheelConstants.LeftMotor.PID_SLOT,
+                NeoFlyWheelConstants.LeftMotor.FEEDFORWARD.calculate(leftVelocity)
+        );
+        
+        
+        SmartDashboard.putNumber("f l v",  NeoFlyWheelConstants.LeftMotor.FEEDFORWARD.calculate(leftVelocity));
```
Comment (itamaroryan) [inline src/main/java/edu/greenblitz/robotName/subsystems/Shooter/FlyWheel/NeoFlyWheel/NeoFlyWheel.java:55]: WTF is an flv?

File: src/main/java/edu/greenblitz/robotName/subsystems/Shooter/FlyWheel/NeoFlyWheel/NeoFlyWheel.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+
+        SmartDashboard.putNumber("left-velocity", leftMotor.getEncoder().getVelocity());
+        SmartDashboard.putNumber("left-current", leftMotor.getOutputCurrent());
+        SmartDashboard.putNumber("left-voltage", leftMotor.getAppliedOutput()  * Battery.getInstance().getCurrentVoltage());
+
+        SmartDashboard.putNumber("right-velocity", rightMotor.getEncoder().getVelocity());
+        SmartDashboard.putNumber("right-current", rightMotor.getOutputCurrent());
+        SmartDashboard.putNumber("right-voltage", rightMotor.getAppliedOutput() * Battery.getInstance().getCurrentVoltage());
```
Comment (itamaroryan) [inline src/main/java/edu/greenblitz/robotName/subsystems/Shooter/FlyWheel/NeoFlyWheel/NeoFlyWheel.java:72]: Remove these

**Review by itamaroryan — CHANGES_REQUESTED** [review]
Noam please clean up the code before PRing next time

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java [hunk 5]
```diff
@@ -5,43 +5,53 @@
...
 
 
 import static edu.greenblitz.robotName.subsystems.shooter.FlyWheel.FalconFlyWheel.FalconFlyWheelConstants.*;
 
 
 public class FalconFlyWheel implements IFlyWheel {
-    private TalonFX motor;
+    private TalonFX rightMotor;
+    private TalonFX leftMotor;
+
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java:16]: spaces

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java [hunk 6]
```diff
@@ -5,43 +5,53 @@
...
+
+    VelocityVoltage rightMotorVelocityVoltage = new VelocityVoltage(0).withEnableFOC(true),
+            leftMotorVelocityVoltage = new VelocityVoltage(0).withEnableFOC(true);
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java:19]: move this or something why is this here

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java [hunk 7]
```diff
@@ -5,43 +5,53 @@
...
+        rightMotor.setNeutralMode(leftMotorConstants.NEUTRAL_MODE_VALUE);
     }
 
     @Override
-    public void setPower(double power) {
-        motor.set(power);
+    public void setPower(double rightPower, double leftPower) {
+        rightMotor.set(rightPower);
+        leftMotor.set(leftPower);
     }
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java:39]: the rightPower is to the left of the leftPower
```suggestion
    public void setPower(double leftPower, double rightPower) {
        rightMotor.set(rightPower);
        leftMotor.set(leftPower);
    }
```

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java [hunk 8]
```diff
@@ -5,43 +5,53 @@
...
 
     @Override
-    public void setVoltage(double voltage) {
-        motor.setVoltage(voltage);
+    public void setVoltage(double rightVoltage, double leftVoltage) {
+        rightMotor.setVoltage(rightVoltage);
+        leftMotor.setVoltage(leftVoltage);
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java:44]: same here

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java [hunk 9]
```diff
@@ -5,43 +5,53 @@
...
     }
 
     @Override
-    public void setVelocity(double velocity) {
-        motor.setControl(new VelocityVoltage(velocity));
+    public void setVelocity(double rightVelocity, double leftVelocity) {
+        rightMotor.setControl(rightMotorVelocityVoltage.withVelocity(rightVelocity));
+        leftMotor.setControl(leftMotorVelocityVoltage.withVelocity(leftVelocity));
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java:50]: double same here

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheel.java [hunk 10]
```diff
@@ -26,16 +26,20 @@ public static FlyWheel getInstance() {
         return instance;
     }
 
-    public void setPower(double power) {
-        flyWheel.setPower(power);
+    public void setPower(double rightPower, double leftPower) {
+        flyWheel.setPower(rightPower, leftPower);
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheel.java:30]: triple same here

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheel.java [hunk 11]
```diff
@@ -26,16 +26,20 @@ public static FlyWheel getInstance() {
...
     }
 
-    public void setVelocity(double velocity) {
-        flyWheel.setVelocity(velocity);
+    public void setVoltage (double rightVoltage, double leftVoltage){
+        flyWheel.setVoltage(rightVoltage,leftVoltage);
+    }
+
+    public void setVelocity(double rightVelocity,double leftVelocity) {
+        flyWheel.setVelocity(rightVelocity, leftVelocity);
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheel.java:38]: quadruple same here

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/IFlyWheel.java [hunk 12]
```diff
@@ -3,11 +3,11 @@
...
 
 public interface IFlyWheel {
-    void setPower(double power);
+    void setPower(double rightPower, double leftPower);
 
-    void setVoltage(double voltage);
+    void setVoltage(double rightVoltage, double leftVoltage);
 
-    void setVelocity(double velocity);
+    void setVelocity(double rightVelocity, double leftVelocity);
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/IFlyWheel.java:10]: quintuple same here

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/NeoFlyWheel/NeoFlyWheel.java [hunk 13]
```diff
@@ -6,40 +6,52 @@
 import edu.greenblitz.robotName.subsystems.shooter.FlyWheel.FlyWheelInputsAutoLogged;
 import edu.greenblitz.robotName.subsystems.shooter.FlyWheel.IFlyWheel;
 import edu.greenblitz.robotName.utils.motors.GBSparkMax;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
 
 public class NeoFlyWheel implements IFlyWheel {
-    private GBSparkMax motor;
+    private GBSparkMax rightMotor, leftMotor;
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/NeoFlyWheel/NeoFlyWheel.java:13]: ok im going to stop here but change all of these

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheel.java [hunk 14]
```diff
@@ -26,16 +26,20 @@ public static FlyWheel getInstance() {
...
     }
 
     public void stop() {
-        flyWheel.setPower(0);
+        flyWheel.setPower(0,0);
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheel.java:42]:
```suggestion
        flyWheel.setPower(0, 0);
```

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheelInputs.java [hunk 15]
```diff
@@ -11,4 +11,5 @@ public class FlyWheelInputs {
     public double temperature;
 
     public double velocity;
+    public double position;
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheelInputs.java:14]: space

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/NeoFlyWheel/NeoFlyWheel.java (hunk 13 above)
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/NeoFlyWheel/NeoFlyWheel.java:13]: space

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/NeoFlyWheel/NeoFlyWheelConstants.java [hunk 16]
```diff
@@ -1,13 +1,31 @@
...
+	public static class LeftMotor{
+		public static final int MOTOR_ID = 22;
+		public static final SimpleMotorFeedforward FEEDFORWARD = new SimpleMotorFeedforward(0.5, 0.0022425, 0.00019209);
+		public static final int PID_SLOT = 0;
+		public static GBSparkMax.SparkMaxConfObject CONFIG = new GBSparkMax.SparkMaxConfObject()
+				.withPID(new PIDObject(100,5,0))
+				.withIdleMode(CANSparkMax.IdleMode.kCoast)
+				.withInverted(false);
+
+	}
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/NeoFlyWheel/NeoFlyWheelConstants.java:29]: space every constant pls

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/simulationFlyWheel/SimulationFlyWheel.java [hunk 17]
```diff
@@ -9,46 +9,66 @@
...
 
 public class SimulationFlyWheel implements IFlyWheel {
-    private FlywheelSim flyWheelSimulation;
-    private double appliedVoltage;
-    private PIDController pidController;
+    private FlywheelSim rightMotorFlywheel;
+    private FlywheelSim leftMotorFlywheel;
+    private double rightMotorApplied, leftMotorApplied;
+    private PIDController rightMotorPIDController;
+    private PIDController leftMotorPIDController;
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/simulationFlyWheel/SimulationFlyWheel.java:16]: spaces

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/simulationFlyWheel/SimulationFlyWheelConstants.java [hunk 18]
```diff
@@ -4,13 +4,28 @@
...
 
-    public static final double GEARING = 8;
+    public static class RightMotor{
+        public static final int NUMBER_OF_MOTORS = 1;
 
-    public static final double J_KG_METERS_SQUARED = 1;
+        public static final double GEARING = 8;
 
-    public static final double POWER_TO_VELOCITY_FACTOR = 0.1;
+        public static final double J_KG_METERS_SQUARED = 1;
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/simulationFlyWheel/SimulationFlyWheelConstants.java:13]: You mean Moment Of Inertia?

**Review by YoniKiriaty — COMMENTED** [review]
right should be right and left should be left

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheelConstants.java [hunk 19]
```diff
@@ -6,13 +6,16 @@
 
 public class FlyWheelConstants {
 
-    public static final double EPSILON_RPM = 50;
+    public static final double EPSILON_RPM = 0;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheelConstants.java:9]: not 0

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/simulationFlyWheel/SimulationFlyWheel.java [hunk 20]
```diff
@@ -9,46 +9,71 @@
...
-    private double appliedVoltage;
-    private PIDController pidController;
+
+    private FlywheelSim rightMotorFlywheel;
+
+    private FlywheelSim leftMotorFlywheel;
+
+    private double rightMotorApplied;
+
+    private double leftMotorApplied;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/simulationFlyWheel/SimulationFlyWheel.java:19]:
```suggestion
    private double rightMotorAppliedOutput;

    private double leftMotorAppliedOutput;
```

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Mechanism/ShooterMechanism.java [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+import static edu.greenblitz.robotName.subsystems.shooter.Mechanism.ShooterMechanismConstants.*;
+import static edu.greenblitz.robotName.subsystems.shooter.Mechanism.ShooterMechanismConstants.PivotMechanismConstants.*;
+
+public class ShooterMechanism extends GBSubsystem {
+
+    private static Mechanism2d shooterMechanism;
+
+    private MechanismRoot2d rootPivot;
+
+    private final MechanismLigament2d pivot;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Mechanism/ShooterMechanism.java:22]:
```suggestion
    private final MechanismLigament2d pivotMechanism;
```

**Review by bossyshrimp68 — DISMISSED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/simulationFlyWheel/SimulationFlyWheel.java (hunk 20 above)
Comment (bossyshrimp68) [reply src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/simulationFlyWheel/SimulationFlyWheel.java:19]: wait why do we even need this here

**Review by roygold8 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/Robot.java [hunk 22]
```diff
@@ -64,7 +63,7 @@ public void initializeSubsystems() {
         Wrist.init();
         Roller.init();
 
-        Lifter.init();
+//        Lifter.init();
```
Comment (itamaroryan) [inline src/main/java/edu/greenblitz/robotName/Robot.java:66]: Don't leave commented code

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java [hunk 23]
```diff
@@ -6,43 +6,55 @@
...
-        motor.getConfigurator().apply(CLOSED_LOOP_RAMPS_CONFIGS);
-        motor.getConfigurator().apply(MOTION_MAGIC_CONFIGS);
-        motor.setNeutralMode(NEUTRAL_MODE_VALUE);
+        rightMotor = new TalonFX(rightMotorConstants.ID);
+        rightMotor.getConfigurator().apply(rightMotorConstants.CURRENT_LIMITS_CONFIGS);
+        rightMotor.getConfigurator().apply(rightMotorConstants.CLOSED_LOOP_RAMPS_CONFIGS);
+        rightMotor.getConfigurator().apply(rightMotorConstants.MOTION_MAGIC_CONFIGS);
+        rightMotor.setNeutralMode(rightMotorConstants.NEUTRAL_MODE_VALUE);
+
+        rightMotor = new TalonFX(leftMotorConstants.ID);
```
Comment (itamaroryan) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FalconFlyWheel/FalconFlyWheel.java:31]: change to left motor ang GBTalonFXPro

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivot.java [hunk 24]
```diff
@@ -11,6 +11,8 @@
 
 import static edu.greenblitz.robotName.subsystems.shooter.Pivot.FalconPivot.FalconPivotConstants.*;
 
+import static edu.greenblitz.robotName.subsystems.shooter.Pivot.FalconPivot.FalconPivotConstants.*;
+import static edu.greenblitz.robotName.subsystems.shooter.Pivot.MotorConstants.SWITCH_CONFIGS;
```
Comment (itamaroryan) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivot.java:15]: Unused imports

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivotConstants.java [hunk 25]
```diff
@@ -1,6 +1,9 @@
 package edu.greenblitz.robotName.subsystems.shooter.Pivot.FalconPivot;
 
 import com.ctre.phoenix6.configs.*;
+import com.ctre.phoenix6.configs.ClosedLoopRampsConfigs;
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.configs.MotionMagicConfigs;
```
Comment (itamaroryan) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivotConstants.java:6]: Unused imports

**Review by itamaroryan — CHANGES_REQUESTED** [review]

**Review by itamaroryan — APPROVED** [review]

**Pull request merged**